docs/build_docs.py

```diff
@@ -0,0 +1,898 @@
+    source_dir,
+    package_name,
+    geoips_docs_dir,
+    build_dir,
+    log=logging.getLogger(__name__),
+):
+    """
+    Set up the documentation source files in the build directory before building.
+
+    This function copies the package documentation source files into a build directory.
+    If the package isn't 'geoips', it setups up template files from the GeoIPS docs.
+    It then creates a customized conf.py and index.rst files tailored for the package.
+
+    Parameters
+    ----------
+    source_dir : str
+        The source directory containing the package's documentation files.
+    package_name : str
+        The name of the package being documented.
+    geoips_docs_dir : str
+        The path to the GeoIPS documentation templates.
+    build_dir : str
+        The directory where the documentation will be built.
+    log : logging.Logger, optional
+        Logger for logging messages; defaults to the module logger.
+    """
+    shutil.copytree(source_dir, build_dir)  # copy docs files over
+
+    if not package_name == "geoips":
+        copy_template_files_to_non_geoips_repo(geoips_docs_dir, build_dir)
+
+    create_conf_py_from_template(build_dir, geoips_docs_dir, package_name)
+
+    required_sections, optional_sections = get_sections()
+    generate_top_level_index_file(
+        build_dir,
+        geoips_docs_dir,
+        package_name,
+        required_sections,
+        optional_sections,
+        log,
+    )
+
+
+def build_module_apidocs_with_sphinx(
+    module_path, apidoc_build_path, log=logging.getLogger(__name__)
+):
+    """
+    Generate API documentation RST files using Sphinx apidoc.
+
+    This function constructs the necessary arguments for sphinx-apidoc to generate
+    the API documentation RST files, excluding certain paths as specified. It then
+    executes sphinx-apidoc and checks for successful completion. It calls the ``main``
+    function of sphinx, and passes arguments to it. This is clunky, but after
+    reviewing the sphinx codebase it seemed to be the best option. Further,
+    there is no (or at least extremely little) documentation on how to use
+    sphinx inside of python; hence the current approach.
+
+    Parameters
+    ----------
+    module_path : str
+        The path to the Python module dir for which to generate API docs.
+    apidoc_build_path : str
+        The output directory where the generated RST files will be placed.
+    log : logging.Logger, optional
+        Logger for logging messages; defaults to the module logger.
+
+    Raises
+    ------
+    Exception
+        If the sphinx-apidoc command fails.
+    """
+    arguments = [
+        "--no-toc",
+        "-o",  # flag for output path
+        apidoc_build_path,  # output path
+        module_path,  # module path
+        "*/lib/*",  # exclude path
+    ]
+    log.debug(f"Running sphinx apidoc with arguments '{' '.join(arguments)}'")
+
+    if not sphinx_apidoc(arguments) == 0:
```

Comment:
This is equivalent to the former command line call?

Reply:
Yes, except failing on warnings